src/clj/game/core/moving.clj

```diff
@@ -497,8 +497,16 @@
                         {:keep-server-alive true
                          :index (card-index state a)
                          :suppress-event true
-                         :swap true})]
+                         :swap true})
+          ;; under the new nsg rules, swapping a card into play triggers an install event
+          ;; TODO - quote exactly where
+          install-event (or (and (installed? a) (not (installed? b)))
+                            (and (installed? b) (not (installed? a))))]
       (trigger-event state side :swap moved-a moved-b)
+      (when (and install-event (= :corp side))
+        (trigger-event state side :corp-install
+                       {:card (get-card state (if (installed? moved-a) moved-a moved-b))
+                        :install-state (:install-state (card-def (if (installed? moved-a) moved-a moved-b)))}))
```

Comment:
This doesn't work and can't work here. `:corp-install` must be async (takes an eid), and this function isn't. Either do the extremely annoying and arduous task of changing all uses of `swap-cards` to be async (might not be possible), or create a new async (takes an eid) function `swap-into-play` that you use in the right card impl.